app/models/media.rb

```diff
@@ -150,17 +150,21 @@ def self.notify_webhook(type, url, data, settings)
         http.use_ssl = uri.scheme == 'https'
         request = Net::HTTP::Post.new(uri.request_uri, headers)
         request.body = payload
-        http.request(request)
+        response = http.request(request)
         @webhook_called = true
+        # .value raises an exception if the response is unsuccessful, but otherwise
+        # returns nil (terribly named method, imo). This is a way to raise that
+        # exception if request failed and still return a successful response if present.
+        response.value || response
       rescue StandardError => e
         PenderAirbrake.notify(e, url: url, type: type, webhook_url: settings['webhook_url'])
         Rails.logger.warn level: 'WARN', message: 'Failed to notify webhook', url: url, type: type, error_class: e.class, error_message: e.message, webhook_url: settings['webhook_url']
-        return false
+        raise(e)
```

Comment:
Will this trigger a report to Errbit? If so, what do you think about deleting line 160?

Reply:
I was wondering the same thing, but wasn't sure and haven't checked locally yet. Going to see now.

Reply:
I think what I actually should do here is make our new exception classes related to things happening to early descend from `Pender::RetryLater` so that they're automatically ignored when the exception is raised, since it will be retried in the future. I'll also delete the Airbrake reporting line on 160!

Reply:
Yes, that sounds great, thanks!